read_spec_dr7.m

```diff
@@ -0,0 +1,43 @@
+% in the same units as the spectrum),
+% the forth row is the mask array.
+%
+% The spectra are binned log-linear. Units are 10^(-17) erg/cm^2/s/Ang
+% http://classic.sdss.org/dr7/products/spectra/read_spSpec.html
+% Notice that the wavelength vector is not contained in the image,
+% but must be generated from parameters in the header. SDSS spectra
+% are binned in constant Log(Λ) and the wavelength can be obtained
+% from the header parameters COEFF0 and COEFF1 (or alternatively
+% CRVAL1 and CD1_1) as follows:
+%
+% Λ = 10**(COEFF0 + COEFF1*i), where i denotes the (zero indexed) pixel number.
+%
+% Remember that these are vacuum wavelengths.
+%
+
+function [wavelengths, flux, noise_variance, pixel_mask] = read_spec_dr7(filename)
+
+measurements = fitsread(filename);
+%           'binarytable',  1, ...
+%           'tablecolumns', 1:4);
+
+% the Primary table contains spectra
+
+% acquire un-continuum subtracted spectrum
+flux = measurements(1,:);
+
+% noise (standard deviation)
+noise = measurements(3,:);
+noise_ratio = noise./flux;
+noise_variance = noise.^2;
+% mask array
+and_mask = measurements(4,:);
+
+% acquire rest wavelength
+header = fitsinfo(filename);
+coef0 = header.PrimaryData.Keywords{19,2}; % Center wavelength (log10) of first pi
+coef1 = header.PrimaryData.Keywords{20,2}; % Log10 dispersion per pixel
```

Comment:
These two keyword locations are incorrect for me, which causes wrong wavelengths with Inf.

For the fits file you gave me, locations for COEF0 and COEF9 are 208 and 209,

```
 coef0 = header.PrimaryData.Keywords{208,2}; % Center wavelength (log10) of first pi
 coef1 = header.PrimaryData.Keywords{209,2}; % Log10 dispersion per pixel
```

Reply:
But I suggest for security, we add an assertion here to prevent some fits files might behave differently:

```
assert(all(header.PrimaryData.Keywords{208,1} == 'COEFF0'))
assert(all(header.PrimaryData.Keywords{209,1} == 'COEFF1'))
```